vertx-web/src/main/java/io/vertx/ext/web/impl/HeaderParser.java

```diff
@@ -0,0 +1,124 @@
+package io.vertx.ext.web.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Build with the intent of following 
+ * <a href="https://tools.ietf.org/html/rfc7231#section-5.3.1">rfc7231,section-5.3.1</a>'s specification.<br>
+ * 
+ */
+public class HeaderParser {
+  
+  
+  private static Pattern COMMA_SPLITTER = Pattern.compile(",(?=(?:(?<!\\\\)\"(?:(?!(?<!\\\\)\").)*(?<!\\\\)\"|\\\\.|[^\"])*$)");
```

Comment:
@pmlopes I raised the strictness of the match for the comma splitter.
It is now aware of `quoted-string` as specified in [rfc7230#section-3.2.6](https://tools.ietf.org/html/rfc7230#section-3.2.6).
Can this be a too large of a performance impact?
If it can, I can move these heavier matches to methods that can "activate" them.